main.nf

```diff
@@ -15,6 +15,22 @@ nextflow.enable.dsl = 2
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
+//Either genome needs to be set or the parameters gtf (for rsem/salmon) and organism, library and keytype
+//(for pathway analysis) have to be provided separately
+//-->gtf is necessary for rsem and salmon
+if (params.input_type in ["rsem", "salmon"]) {
```

Comment:
you add other parameter checks in the `rnadeseq.nf`. Is there a reason to add those here and not also there?

Reply:
Ah, so in rnaseq, the genome params are defined in the main.nf, so I did the same here (but kept the overall input_type check in rnadeseq.nf as usually, the validity checks are done there I think). Should I just move all genome params to rnadeseq.nf or should I remove the input_type NOT IN [rsem, salmon, featurecounts] check to main.nf?

Reply:
ok I see, in that case you can leave it like it is, in the same way that `rnaseq` implements it